src/test/java/teammates/test/cases/datatransfer/AdminEmailAttributesTest.java

```diff
@@ -22,37 +22,121 @@
     private List<String> groupReceiverListFileKey = Arrays.asList("listfilekey", "listfilekey");
     private String subject = "subject of email";
     private Text content = new Text("valid email content");
     private Date date = new Date();
-    private AdminEmailAttributes adminEmailAttributes = new AdminEmailAttributes(
-            subject, addressReceiverListString, groupReceiverListFileKey, content, date);
+    private AdminEmailAttributes validAdminEmailAttributesObject = AdminEmailAttributes
+            .builder(subject, addressReceiverListString, groupReceiverListFileKey, content)
+            .withSendDate(date)
+            .build();
 
     @Test
-    public void testValidate() throws Exception {
+    public void testBuilderWithDefaultValues() {
+        ______TS("valid admin email");
+
+        AdminEmailAttributes attributesWithDefaultValues = AdminEmailAttributes
+                .builder(subject, addressReceiverListString, groupReceiverListFileKey, content)
+                .build();
+
+        ______TS("success: default values for optional params");
+
+        assertEquals(Const.ParamsNames.ADMIN_EMAIL_ID, attributesWithDefaultValues.getEmailId());
+        assertFalse("Default false for isInTrashBin", attributesWithDefaultValues.isInTrashBin);
+        assertEquals(Const.TIME_REPRESENTS_DEFAULT_TIMESTAMP, attributesWithDefaultValues.getCreateDate());
+        assertTrue("Valid input", attributesWithDefaultValues.isValid());
+    }
+
+    @Test
+    public void testBuilderWithNullOptionalArguments() {
+        AdminEmailAttributes attributesWithNullOptionalArguments = AdminEmailAttributes
+                .builder(subject, addressReceiverListString, groupReceiverListFileKey, content)
+                .withSendDate(null)
+                .withCreateDate(null)
+                .withEmailId(null)
+                .build();
+
         ______TS("valid admin email");
 
+        assertTrue("Valid input", attributesWithNullOptionalArguments.isValid());
+
+        ______TS("success: default values for optional params");
+
+        assertEquals(Const.ParamsNames.ADMIN_EMAIL_ID, attributesWithNullOptionalArguments.getEmailId());
+        assertFalse("Default false for isInTrashBin", attributesWithNullOptionalArguments.isInTrashBin);
+        assertEquals(Const.TIME_REPRESENTS_DEFAULT_TIMESTAMP, attributesWithNullOptionalArguments.getCreateDate());
+        assertEquals(null, attributesWithNullOptionalArguments.getSendDate());
+    }
+
+    @Test(expectedExceptions = AssertionError.class, expectedExceptionsMessageRegExp = Const.StatusCodes.NULL_PARAMETER)
+    public void testBuilderWithNullRequiredArguments() {
+        ______TS("failure: subject cannot be null)");
+
+        AdminEmailAttributes
+                .builder(null, addressReceiverListString, groupReceiverListFileKey, content)
+                .build();
+
+        ______TS("failure: addressReceiverListString cannot be null)");
```

Comment:
I don't think this is a good idea; the first instance of `*Exception` thrown will already make this test pass, which means everything past this line will not be executed.